src/communication/communication_server_channel.cpp

```diff
@@ -29,10 +29,11 @@
 namespace cubcomm
 {
 
-  server_channel::server_channel (const char *server_name, int max_timeout_in_ms)
+  server_channel::server_channel (const char *server_name, SERVER_TYPE server_type, int max_timeout_in_ms)
     : channel (max_timeout_in_ms),
       m_server_name (server_name)
   {
+    m_server_type = server_type;
     assert (server_name != NULL);
```

Comment:
new code in hpp/cpp should use `nullptr` all over; macro `NULL` remains for h/c files